retriever/lib/tools.py

```diff
@@ -51,17 +51,31 @@ def create_home_dir():
 
 
 def name_matches(scripts, arg):
+    """Check for a match of the script in available scripts
+
+    if all, return the entire script list
+    if the exact script is available, return that script
+    if no exact script name detected, match the argument with keywords
+    title and name of all scripts and return the closest matches
+    """
     matches = []
+
+    if arg.strip().lower() == "":
```

Comment:
We could probably just set `arg = arg.strip().lower()` immediately before this line and then use `arg` throughout.